src/Surfnet/Stepup/Helper/UserDataFormatter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright 2022 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Helper;
+
+class UserDataFormatter implements UserDataFormatterInterface
+{
+    private $applicationName;
+
+    public function __construct(string $applicationName)
+    {
+        $this->applicationName = $applicationName;
+    }
+
+    public function format(array $userData, array $errors): array
```

Comment:
You could return a simple response DTO to detect breaking changes early on when changing code without introducing side effects and being unaware of this.

Reply:
That was one of the things @VadimSchmitz and I discussed in our whiteboard session. In the end I felt the added benefit you describe was not quite worth the added 'complexity'. I will reconsider our design decisions, and perhaps update the code accordingly.